golang-ci/pipeline.yaml

```diff
@@ -0,0 +1,30 @@
+steps:
+  - label: ":golang: Install"
+    key: "deps"
+    command: "go install"
+    plugins:
+      - docker#v5.9.0:
+          image: "golang:latest"
+
+  - label: ":golang: Build"
+    command: "go build main.go"
+    depends_on: ["deps"]
+    plugins:
+      - docker#v5.9.0:
+          image: "golang:latest"
+
+  - label: ":golang: Test"
+    command: "go test ."
+    depends_on: ["deps"]
+    plugins:
+      - docker#v5.9.0:
+          image: "golang:latest"
+
+  - label: ":golang: Go generate check"
+    commands:
+      - go generate ./...
+      - git diff --exit-code || (echo 'Generated code is out of date, please run `go generate` and commit the changes' && exit 1)
```

Comment:
❤️

Reply:
Just had a quick test of this locally and something to note is that `git diff` doesn't play nicely with the local run since there's no git repository present. 

ie.
```
cd example-project; bk local run ../pipeline.yaml
```

I think it's fine, just something to be aware of. I'd hazard a guess we'll run into this again in the future so perhaps having an easier entry point script that does something to bring in the git repo is worth considering. I'll pop that in the laterbase.